### ISSUE TITLE
[Toolkit feature]: Allow distant lights to shine through select meshes

### ISSUE THREAD

**Issue by PappaSlask** [NVIDIAGameWorks/rtx-remix#475]
### What is the context of the feature/improvement?

I was adding in a skydome over my level, and wanted to add a simple distant light to act as a sun. But the light was blocked by the dome mesh and couldn't lit my scene. Please give us a way to set certain meshes to allow lights to pass through without any loss. I know something similar can be done via subsurface scattering, but it wont allow 100% of the light through, and it will make the mesh itself bright, which I don't want for my skydome.

### Please describe the feature/improvement?

As described above

### What would be your solution? (optional)

...

### Version

2024.3.0

**Comment by nvdamien** [comment]
Hi,

I believe this is a rendering thing (pass through geometry).

In any case, anything wrong to use a point light?

**Comment by PappaSlask** [comment]
> Hi,
> 
> I believe this is a rendering thing (pass through geometry).
> 
> In any case, anything wrong to use a point light?

Indeed, I guess it's called something like that. Well, there's always the option to bring in a point light. But there are several issues with that compared to having a dedicated distant light:
- You would need to place it carefully to mimic the correct distance, to get the cone, radius, softness right, it also affects how the light interacts with volumetric lighting
- The settings has to be tweaked carefully for the same reasons mentioned above
- Adding a sphere light on the inside of your dome would make light bounce and hit the inside of the dome walls, this could maybe be countered by adjusting the cone settings, but wouldn't be an optimal solution.

**Comment by NV-LL** [comment]
Hi @PappaSlask - what game are you working on that you'd like this feature for?

**Comment by NV-LL** [comment]
@PappaSlask checking back in to see what game you're working on and would like this feature for - that context will help us better understand the request. We also want to inform you of our stale ticket policy, namely that we will close this as 'went-stale' if there is no update in the next two weeks.

**Comment by NV-LL** [comment]
Hi @PappaSlask - we're going to close this ticket as 'went stale' as it hasn't received an update in the past two weeks, per the notice above. If you are still actively pursuing this feature and can provide us with the requested info, please reopen this ticket and let us know. Thanks!

**Comment by PappaSlask** [comment]
@NV-LL Again, I'm sorry about the delays. The game in question that I'm working on is Max Payne (2001), and it relies on skydomes for their skies, basically a half-sphere mesh with a sky texture slapped onto it. If you guys could engineer a way to let a distant light (or any light really) pass through the dome mesh, it would be super cool. 

It would also be useful in alotta other games as well, especially since the fallback light is set to Distant by default, which means that whenever users jump into a new game which uses any form of skybox, be it a dome, sphere or a square, that fallback light <u>will</u> get blocked and won't do what it's supposed to do, and the scene will be pitch black in many games.

**Comment by nsubtil** [comment]
I think the intent here is that you'd be able to reuse the original game's sky geo, but replace textures on it. The reason is that if you tag the incoming geo from the game as sky, we'll bake it down to an environment map on the fly, which lets distant lights work.

Is there a particular reason you're explicitly adding a sky dome instead of reusing the original sky dome from the game?

**Comment by NV-LL** [comment]
Hi @PappaSlask - checking in to make sure you saw the question in the above comment. We don't want to let this ticket go stale if you do intend to keep pursuing a solution. Thanks!
